jazz_scm-webhook/index.js

```diff
@@ -0,0 +1,180 @@
+		var servContext = getServiceContext(eventKey, eventBody),		
+			bodyObj = {
+				'event_handler': config.event_handler.bitbucket,
+				'event_name': servContext.event_name, 
+				'service_name': serviceName,
+				'event_status': config.event_status.completed,
+				'event_type': servContext.event_type, 
+				'username': userName,
+				'event_timestamp': timestamp,
+				'service_context': {      
+					'repository': repositoryLink,
+					'domain' : domain,
+					'branch' : servContext.branch,
+					'pr_link' : servContext.prlink,
+					'target' : servContext.target,
+					'hash' : servContext.hash
+				}
+			};
+		var possibleEventName = config.event_name,
+			isvalidEventName = false;
+		for (var idx in possibleEventName ){
+			if ((servContext.event_name).toLowerCase() == possibleEventName[idx].toLowerCase()){
+				isvalidEventName = true;
+			} 
+		}
+		
+		logger.info("bodyObj::" +JSON.stringify(bodyObj));
+		logger.info("config.events_url::" + config.events_url);
+
+		if(isvalidEventName){
+			//Invoking platform events api to record bit bucket activity.	
+			var options = {
+				url: config.events_url,
+				method: 'POST',
+				headers: { 
+					'cache-control': 'no-cache',
+					'content-type': 'application/json' 
+				},
+				body : bodyObj,
+				json: true,
+				rejectUnauthorized: false
+			};
+			request(options, function (error, response, body) {
+				if (error) {
+					logger.error('Error invoking service: ' + error);
+					return cb(JSON.stringify(errorHandler.throwInternalServerError("Unexpected Error occured,"+ error)));
+				} else {
+					if(response.statusCode === 200){
+						var output = {
+							message: 'successfully recorded bitbucket activity to jazz_events.',
+							event_id : body.data.event_id
+						};
+						logger.info("OutPut: " +JSON.stringify(output));
+						logger.info("Input: " +JSON.stringify(body.input));
+						cb(null, responseObj(output, body.input));
+					} else {
+						logger.error("StatusCode :"+ response.statusCode);
+						return cb(JSON.stringify(errorHandler.throwInternalServerError("Unable to send events, StatusCode:"+ response.statusCode)));
+					}
+				}
+			});
+		} else {
+			logger.warn("Unable to send envents! Only specified event name can be allowed :" + possibleEventName.join(", "));
+		}
+		
+	} else {
+		return cb(JSON.stringify(errorHandler.throwInternalServerError("Eventkey is null or undefined! so unable to find event name.")));
+	}	
+	
+	function getServiceContext(value, body){
+		var result = {},
+			changes = null;
+		result.event_type = config.event_type.deployment;
+		if(body.pullrequest){
+			changes = body.pullrequest;	
+			result.branch = changes.fromRef.branch.name;
+			result.prlink = body.pullrequest.link;
+			result.target = changes.toRef.branch.name;
+		} else if(body.push){
+			changes = body.push.changes[0];	
+		}
+		
+		switch (value) {
+			case 'repo:push': 					
+				if(changes.created && !changes.closed && changes.old === null && changes.new !== null) {
```

Comment:
Check if changes is not null. Also combine and use short form:
`if(changes && changes.created && !changes.closed && !changes.old && changes.new) {`

Reply:
Done